src/encoder/init.cpp

```diff
@@ -31,7 +31,12 @@ NAN_METHOD(encodeToPngBuffer) {
     int compression = args[3].As<Integer>()->Value();
     bool interlaced = args[4]->BooleanValue();
     bool trans = args[5]->BooleanValue();
-    NanCallback * callback = new NanCallback(args[6].As<Function>());
+
+    int metadata_len = args[6].As<String>()->Utf8Length();
```

Comment:
Does this include the null termination character? I think it does, but I just want to make sure we allocate enough space for the string.